dask/dataframe/backends.py

```diff
@@ -213,6 +215,18 @@ def get_pyarrow_schema_pandas(obj):
     return pa.Schema.from_pandas(obj)
 
 
+@to_pyarrow_table_dispatch.register((pd.DataFrame,))
+def get_pyarrow_table_from_pandas(obj, preserve_index=True):
+    import pyarrow as pa
+
+    return pa.Table.from_pandas(obj, preserve_index=preserve_index)
+
+
+@from_pyarrow_table_dispatch.register((pd.DataFrame,))
+def get_pandas_dataframe_from_pyarrow(_, table, **kwargs):
```

Comment:
I'm wondering how useful a dispatch mechanism with `**kwargs` would be. I suspect that keyword arguments will heavily depend on the dispatch type, defeating the purpose of the dispatch. What are your thoughts?

Reply:
Great question. You are correct that we typically try to formalize the set of key-word arguments that a dispatch implementation is expected to support. For this particular case, cudf doesn't accept any key-word arguments at all, while pandas accepts [many](https://arrow.apache.org/docs/python/generated/pyarrow.Table.html#pyarrow.Table.to_pandas).

The challenge in this particular case is that the p2p algorithm now uses `self_destruct=True`, which is not supported by cudf. Perhaps we can just use our own `allow_self_destruct` argument to imply that the self destruct support is optional?

Reply:
I don't think that's blocking this PR though, we can iterate on the dispatch mechanism depending on future needs.

Reply:
Some thoughts on this: 

* The value of `self_destruct=True` is not yet proven. We could take a look at whether having it makes a difference and remove it if not.

* If it does make a difference, we could add an `allow_self_destruct` keyword that `cudf` ignores or rename this dispatch something like `move_from_pyarrow_table_dispatch` or `destructive_from_pyarrow_table_dispatch` (cudf once again would simply ignore the destructiveness) to signal that this dispatch is (potentially) destructive.

Reply:
Also, if we're being pedantic here, `self_destruct` only _attempts_ to deallocate the originating Arrow memory while converting the Arrow object to pandas. So if it wouldn't do anything in `cudf`, that technically feels fine to me. To me, `self_destruct=True` is basically just signalling that it's not safe to use the table anymore.

Reply:
So maybe we could even just stick with `self_destruct`, and registered dispatches are allowed to ignore the keyword?

Reply:
Thanks @hendrikmakait - I think I agree with your thoughts.

Reply:
Would you like to change `**kwargs` to be `self_destruct=False` for this PR then? (seems reasonable to me)